public/app/features/integrated-alerting/components/AlertRules/AddAlertRuleModal/AddAlertRuleModal.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Form, Field } from 'react-final-form';
+import { Button, HorizontalGroup, Switch, Select, MultiSelect, useStyles } from '@grafana/ui';
+import {
+  Modal,
+  LoaderButton,
+  TextInputField,
+  NumberInputField,
+  TextareaInputField,
+  logger,
+  validators,
+} from '@percona/platform-core';
+import { SelectableValue } from '@grafana/data';
+import { AppEvents } from '@grafana/data';
+import { Messages } from './AddAlertRuleModal.messages';
+import { AddAlertRuleModalProps, AddAlertRuleFormValues } from './AddAlertRuleModal.types';
+import { getStyles } from './AddAlertRuleModal.styles';
+import { SEVERITY_OPTIONS, NOTIFICATION_CHANNEL_OPTIONS } from './AddAlertRulesModal.constants';
+import { formatTemplateOptions, formatCreateAPIPayload } from './AddAlertRuleModal.utils';
+import { AlertRulesService } from '../AlertRules.service';
+import { AlertRuleTemplateService } from '../../AlertRuleTemplate/AlertRuleTemplate.service';
+import { appEvents } from 'app/core/core';
+
+const { required } = validators;
+
+export const AddAlertRuleModal: FC<AddAlertRuleModalProps> = ({ isVisible, setVisible }) => {
+  const styles = useStyles(getStyles);
+  const [templateOptions, setTemplateOptions] = useState<Array<SelectableValue<string>>>();
+
+  const getTemplates = async () => {
+    try {
+      const response = await AlertRuleTemplateService.list();
+      setTemplateOptions(formatTemplateOptions(response.templates));
+    } catch (e) {
+      logger.error(e);
+    }
+  };
+
+  useEffect(() => {
+    getTemplates();
+  }, []);
+
+  const onSubmit = async (values: AddAlertRuleFormValues) => {
+    try {
+      await AlertRulesService.create(formatCreateAPIPayload(values));
+      setVisible(false);
+      appEvents.emit(AppEvents.alertSuccess, [Messages.addSuccess]);
+    } catch (e) {
+      logger.error(e);
+    }
+  };
+
+  return (
+    <Modal
+      title={Messages.title}
+      isVisible={isVisible}
+      onClose={() => setVisible(false)}
+      data-qa="add-alert-rule-modal"
+    >
+      <Form
+        onSubmit={onSubmit}
+        render={({ handleSubmit, valid, pristine, submitting }) => (
+          <form className={styles.form} onSubmit={handleSubmit} data-qa="add-alert-rule-modal-form">
+            {/* TODO: polish this up */}
+            <dl className={styles.formFieldsWrapper}>
+              <dt>{Messages.templateField}</dt>
+              <dd>
+                <Field name="template" validate={required}>
+                  {({ input }) => (
+                    <Select className={styles.select} options={templateOptions} {...input} data-qa="template-select" />
+                  )}
+                </Field>
+              </dd>
+
+              <dt>{Messages.nameField}</dt>
```

Comment:
Why not use the existing label for some fields that have it? 🤔

Reply:
Because it's not on the same line, it will be shown on top of the component, which is different from what we have in the design

Reply:
I think we should be consistent and keep it on the top. At least that's what Im doing even though it's not whats on the design. What do you think? @atymchuk

Reply:
@tiagomotasantos 
Yes, we should stick with keeping labels on top of the fields. And we even have it as a requirement in our docs (internal, so I'll post a link on slack).

Reply:
The design need to be updated then. I'll use the labels on the top

Reply:
Done